annotations/s2i-annotations/src/main/java/io/dekorate/s2i/decorator/AddDockerImageStreamResourceDecorator.java

```diff
@@ -36,7 +38,9 @@ public AddDockerImageStreamResourceDecorator(ImageConfiguration config, String d
   }
 
   public void visit(KubernetesListBuilder list) {
-    ObjectMeta meta = getMandatoryDeploymentMetadata(list);
+    List<ObjectMeta> metaList = getDeploymentMetadataList(list, this.config.getName());
+    //If there are multiple deployment resource we can't possibly determine which one to use.
+    ObjectMeta meta = metaList.size() == 1 ? metaList.get(0) : new ObjectMeta();
```

Comment:
These three lines are equivalent to simply do:
`getDeploymentMetadata(list, this.config.getName()).orElseGet(ObjectMeta::new);`

Reply:
This comment was marked as resolved, but it was not updated. Did you do it on purpose?